flow/connectors/clickhouse/normalize.go

```diff
@@ -149,19 +149,16 @@ func (c *ClickhouseConnector) NormalizeRecords(req *model.NormalizeRecordsReques
 			ct := column.Type
 
 			colSelector.WriteString(fmt.Sprintf("%s,", cn))
-			// if i < numCols-1 {
-			// 	colSelector.WriteString(",")
-			// }
-
-			extractionFuction := "JSONExtractRaw"
-			switch qvalue.QValueKind(ct) {
-			case qvalue.QValueKindString:
-				extractionFuction = "JSONExtractString"
-			case qvalue.QValueKindInt64:
-				// TODO check if int64 is supported.
-				extractionFuction = "JSONExtractInt"
+			colType := qvalue.QValueKind(ct)
+			clickhouseType, err := qValueKindToClickhouseType(colType)
+			if err != nil {
+				return nil, fmt.Errorf("error while converting column type to clickhouse type: %w", err)
 			}
-			projection.WriteString(fmt.Sprintf("%s(_peerdb_data, '%s') AS %s, ", extractionFuction, cn, cn))
+			if clickhouseType == "DateTime64(6)" {
+				clickhouseType = "String"
+			}
+
+			projection.WriteString(fmt.Sprintf("JSONExtract(_peerdb_data, '%s', '%s') AS %s, ", cn, clickhouseType, cn))
```

Comment:
quote `cn`

Reply:
Is it just a hygiene or any specific reason for this. Column names cannot have space or anything right

Reply:
[Yes, they can.](https://www.postgresql.org/docs/7.0/syntax525.htm#:~:text=Names%20containing%20other%20characters%20may,always%20folded%20to%20lower%20case.) Our codebase has generally relied on customers keeping plain names, at least without internal quotes, but this is something we need to fix

Reply:
clickhouse identifier syntax: https://clickhouse.com/docs/en/sql-reference/syntax#identifiers